src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -547,7 +547,7 @@ private static function parseUrl(string $url, array $query = [], array $allowedS
             }
 
             // https://tools.ietf.org/html/rfc3986#section-3.3
-            $parts[$part] = preg_replace_callback("#[^-A-Za-z0-9._~!$&/'()[\]*+,;=:@\\\\^`{|}%]++#", function ($m) { return rawurlencode($m[0]); }, $parts[$part]);
+            $parts[$part] = preg_replace_callback("#[^-A-Za-z0-9._~!$&/'()[\]*+,;=:@?\#%]++#", function ($m) { return rawurlencode($m[0]); }, $parts[$part]);
```

Comment:
This force-encodes characters that are not in the list.
Which means with this change, `?` and `#` won't be encoded. This looks incorrect to me (even if I don't expect these to be found in parts of valid URLs).
Also *when the original URL contain these unencoded*, there is on need to force-encode the chars removed in the diff.
Which means I would keep the current regexp.

Reply:
Ok can we remove `|` char from original regexp, this one is the problematic one, which cause issues for us.

Reply:
And I still do not understand why the original PR https://github.com/symfony/symfony/pull/49299 has been merged to 5.4 branch? Maybe rever it from 5.4 branch would be ok too?